services/api-db/docker-entrypoint-initdb.d/01-migrations.sql

```diff
@@ -1772,6 +1772,706 @@ CREATE OR REPLACE PROCEDURE
+        AND table_schema = 'infrastructure'
+        AND column_name = 'environment'
+    ) THEN
+      ALTER TABLE `environment_service`
+      MODIFY environment int NOT NULL;
+    END IF;
+  END;
+$$
+
+CREATE OR REPLACE PROCEDURE
+  drop_foreign_key_from_task()
+
+  BEGIN
+    IF EXISTS (
+      SELECT *
+      FROM INFORMATION_SCHEMA.REFERENTIAL_CONSTRAINTS
+      WHERE
+        CONSTRAINT_SCHEMA = 'infrastructure'
+        AND REFERENCED_TABLE_NAME = 'environment'
+        AND CONSTRAINT_NAME = 'task_ibfk_1'
+    ) THEN
+      ALTER TABLE `task`
+        MODIFY environment int NOT NULL, DROP FOREIGN KEY task_ibfk_1;
+    ELSEIF EXISTS (
+    	SELECT NULL
+		  FROM INFORMATION_SCHEMA.COLUMNS
+		  WHERE
+        table_name = 'task'
+        AND table_schema = 'infrastructure'
+        AND column_name = 'environment'
+    ) THEN
+      ALTER TABLE `task`
+      MODIFY environment int NOT NULL;
+    END IF;
+  END;
+$$
+
+CREATE OR REPLACE PROCEDURE
+  drop_foreign_key_from_environment_problem()
+
+  BEGIN
+    IF EXISTS (
+      SELECT *
+      FROM INFORMATION_SCHEMA.REFERENTIAL_CONSTRAINTS
+      WHERE
+        CONSTRAINT_SCHEMA = 'infrastructure'
+        AND REFERENCED_TABLE_NAME = 'environment'
+        AND CONSTRAINT_NAME = 'environment_problem_ibfk_1'
+    ) THEN
+      ALTER TABLE `environment_problem`
+        MODIFY environment int, DROP FOREIGN KEY environment_problem_ibfk_1;
+    ELSEIF EXISTS (
+    	SELECT NULL
+		  FROM INFORMATION_SCHEMA.COLUMNS
+		  WHERE
+        table_name = 'environment_problem'
+        AND table_schema = 'infrastructure'
+        AND column_name = 'environment'
+    ) THEN
+      ALTER TABLE `environment_problem`
+      MODIFY environment int;
+    END IF;
+  END;
+$$
+
+CREATE OR REPLACE PROCEDURE
+  drop_foreign_key_from_project_notification()
+
+  BEGIN
+    IF EXISTS (
+      SELECT *
+      FROM INFORMATION_SCHEMA.REFERENTIAL_CONSTRAINTS
+      WHERE
+        CONSTRAINT_SCHEMA = 'infrastructure'
+        AND REFERENCED_TABLE_NAME = 'project'
+        AND CONSTRAINT_NAME = 'project_notification_ibfk_1'
+    ) THEN
+      ALTER TABLE `project_notification`
+      MODIFY pid int, DROP FOREIGN KEY project_notification_ibfk_1;
+    ELSEIF EXISTS (
+    	SELECT NULL
+		  FROM INFORMATION_SCHEMA.COLUMNS
+		  WHERE
+        table_name = 'project_notification'
+        AND table_schema = 'infrastructure'
+        AND column_name = 'pid'
+    ) THEN
+      ALTER TABLE `project_notification`
+      MODIFY pid int;
+    END IF;
+  END;
+$$
+
+CREATE OR REPLACE PROCEDURE
+  drop_foreign_key_usid_from_user_ssh_key()
+
+  BEGIN
+    IF EXISTS (
+      SELECT *
+      FROM INFORMATION_SCHEMA.REFERENTIAL_CONSTRAINTS
+      WHERE
+        CONSTRAINT_SCHEMA = 'infrastructure'
+        AND REFERENCED_TABLE_NAME = 'user'
+        AND CONSTRAINT_NAME = 'user_ssh_key_ibfk_1'
+    ) THEN
+      ALTER TABLE `user_ssh_key`
+      MODIFY usid int, DROP FOREIGN KEY user_ssh_key_ibfk_1;
```

Comment:
the usid here is already a `char(36)` from an earlier migration

Reply:
Good catch! Updated now